relayer/relays/beacon/main.go

```diff
@@ -269,13 +291,38 @@ func (r *Relay) SyncHeaders(ctx context.Context) error {
 		blockRoot = common.HexToHash(headerUpdate.Block.ParentRoot.Hex())
 		prevSyncAggregate = headerUpdate.Block.Body.SyncAggregate
 	}
-	
 
 	// Import the execution header for the second last finalized header too.
 	_, err = r.SyncHeader(ctx, blockRoot, prevSyncAggregate)
 	if err != nil {
 		return err
 	}
 
+	lastBlockNumber, secondLastBlockNumber, err := r.syncer.GetBlockRange(lastFinalizedHeader, secondLastFinalizedHeader)
+	if err != nil {
+		return err
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"start": secondLastBlockNumber,
+		"end":   lastBlockNumber - 1,
+	}).Info("Processing events for block numbers")
+
+	payload, err := r.listener.ProcessEvents(ctx, secondLastBlockNumber, lastBlockNumber-1)
```

Comment:
Looking over the relayer code reminds me that we still need to handle certain recovery scenarios (in a later PR). Suppose the relayer dies somehow after importing an execution header into the parachain, but before importing the events for that execution header. Then when the relayer starts up it doesn't know that it still needs to import some old events, and moves onto the next header.

So we need to digest the problem and figure out a solution. In a later PR of course.

One idea is for the parachain inbound channels to keep track of the blockNumber of the last imported event. Then when the relayer startup it queries that block number, and begins relaying ethereum events from that block number. Furthermore, messages from ethereum are basically events like `Message(nonce, payload)`. So the relayer can inspect the message nonce to see if it is the next message expected by the inbound channel pallets in the Parachain.

I'm sure there other solutions too.

Reply:
I logged Linear issue [SNO-292](https://linear.app/snowfork/issue/SNO-292/relayer-and-beacon-client-should-support-syncing-from-last-synced-slot) for this.